File: static/java/FormulaLookupField.java
// ==========================================
// MyAirtable static runtime.
// ==========================================

package myairtable;

/**
 * Formula builder for {@code multipleLookupValues} / lookup fields.
 *
 * <p>Lookup field values are arrays. Airtable does not auto-coerce arrays through {@code LOWER} /
 * {@code TRIM} / {@code FIND}, so {@code contains}, {@code startsWith} and {@code endsWith} would
 * silently match nothing. {@link #stringField} wraps the reference in {@code ARRAYJOIN(field, ",
 * ")} so string ops see a string.
 *
 * <p>Equality ({@code =}) is unaffected — Airtable coerces array fields to a comma-joined string
 * under {@code =}, so direct equality already works.
 */
public final class FormulaLookupField implements FormulaTextOps {

  private final String fieldId;

  public FormulaLookupField(String fieldId) {
    this.fieldId = fieldId;
  }

  @Override
  public String getFieldId() {
    return fieldId;
  }

  @Override
  public String stringField() {
    return "ARRAYJOIN(" + field() + ", \", \")";
  }
}

File: src/myairtable/static/csharp/FormulaLookupField.cs
namespace MyAirtable;

/// <summary>
/// Formula builder for <c>multipleLookupValues</c> / lookup fields.
///
/// <para>Lookup field values are arrays. Airtable does not auto-coerce arrays through LOWER / TRIM /
/// FIND, so Contains/StartsWith/EndsWith would silently match nothing. <see cref="StringField"/>
/// wraps the reference in <c>ARRAYJOIN(field, ", ")</c> so string ops see a string. Equality (=) is
/// unaffected — Airtable coerces array fields to a comma-joined string under =.</para>
/// </summary>
public sealed class FormulaLookupField : FormulaTextOps
{
    public FormulaLookupField(string fieldId)
        : base(fieldId) { }

    protected override string StringField() => "ARRAYJOIN(" + Field() + ", \", \")";
}
